### PR TITLE
use Actions across the board

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
Windows is failing due to using `\r\n` instead of `\n` in some places.

**Comment by dpo** [comment]
Is there a smarter string comparison function that would take that into account?

File: test/pkgbmark.jl [hunk 1]
```diff
@@ -4,6 +4,8 @@ using PkgBenchmark
 import Plots
 
 function test_pkgbmark()
+  repo = LibGit2.GitRepo(joinpath(@__DIR__, ".."))
```
Comment (dpo) [inline test/pkgbmark.jl:7]:
```suggestion
```

File: test/pkgbmark.jl [hunk 2]
```diff
@@ -4,6 +4,8 @@ using PkgBenchmark
...
+  head_oid = LibGit2.head_oid(repo)
```
Comment (dpo) [inline test/pkgbmark.jl:8]:
```suggestion
```

File: test/pkgbmark.jl [hunk 3]
```diff
@@ -21,8 +23,8 @@ function test_pkgbmark()
   p = profile_solvers(stats, costs, ["time", "memory", "gctime+1", "allocations"])
   @test typeof(p) <: Plots.Plot
 
-  repo = LibGit2.GitRepo(joinpath(@__DIR__, ".."))
   LibGit2.lookup_branch(repo, "master") === nothing && LibGit2.branch!(repo, "master", force=true)
+  LibGit2.isdirty(repo) && LibGit2.reset!(repo, head_oid, LibGit2.Consts.RESET_HARD)
```
Comment (dpo) [inline test/pkgbmark.jl:27]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -15,11 +15,18 @@ jobs:
...
           - ubuntu-latest
+          - macOS-latest
+          - windows-latest
         arch:
           - x64
     steps:
+      - name: Set git to use LF
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
```
Comment (dpo) [inline .github/workflows/ci.yml:29]:
```suggestion
          git config --global core.eol lf
      if: matrix.os == 'windows-latest'
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -15,11 +15,19 @@ jobs:
...
+      if: matrix.os == 'windows-latest'
```
Comment (dpo) [inline .github/workflows/ci.yml:30]:
```suggestion
        if: matrix.os == 'windows-latest'
```

File: test/pkgbmark.jl [hunk 6]
```diff
@@ -21,7 +21,6 @@ function test_pkgbmark()
   p = profile_solvers(stats, costs, ["time", "memory", "gctime+1", "allocations"])
   @test typeof(p) <: Plots.Plot
 
-  repo = LibGit2.GitRepo(joinpath(@__DIR__, ".."))
   LibGit2.lookup_branch(repo, "master") === nothing && LibGit2.branch!(repo, "master", force=true)
```
Comment (dpo) [inline test/pkgbmark.jl:24]:
```suggestion
  repo = LibGit2.GitRepo(joinpath(@__DIR__, ".."))
  LibGit2.lookup_branch(repo, "master") === nothing && LibGit2.branch!(repo, "master", force=true)
```

**Pull request closed without merging**